dev/com.ibm.ws.persistence/src/com/ibm/wsspi/persistence/internal/DatabaseStoreImpl.java

```diff
@@ -738,6 +781,52 @@ else if (suspendedLTC != null)
         return dbProductName;
     }
 
+    /**
+     * Attempt to ascertain if the given isolation level is supported on the given datasource
+     *<p>
+     *{@link java.sql.DatabaseMetaData#supportsTransactionIsolationLevel()}
+     *
+     * @param dataSource data source instance.
+     * @param isolationLevel valid isolation level value.
+     * @return 'true' if the driver DatabaseMetaData indicates the given transaction isolation is supported; 'false' otherwise
+     */
+    private boolean supportsIsolationLevel(WSDataSource dataSource, int isolationLevel) {
+        try {
+            // Query the metadata under a new transaction and commit right away
+            LocalTransactionCurrent localTranCurrent = this.localTranCurrent;
+            LocalTransactionCoordinator suspendedLTC = localTranCurrent.suspend();
+            EmbeddableWebSphereTransactionManager tranMgr = this.tranMgr;
+            Transaction suspendedTran = suspendedLTC == null ? tranMgr.suspend() : null;
+            boolean tranStarted = false;
+            try {
+                tranMgr.begin();
+                tranStarted = true;
+                Connection con = dataSource.getConnection();
+                try {
+                    return con.getMetaData().supportsTransactionIsolationLevel(isolationLevel);
+                } finally {
+                    con.close();
+                }
+            } finally {
+                try {
+                    if (tranStarted)
+                        tranMgr.commit();
+                } finally {
+                    // resume
+                    if (suspendedTran != null)
+                        tranMgr.resume(suspendedTran);
+                    else if (suspendedLTC != null)
+                        localTranCurrent.resume(suspendedLTC);
+                }
+            }
+           } catch (Exception e) {
+               if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
```

Comment:
Is this something we might want to FFDC?  If something's going wrong here in the field, we won't know until trace is turned on.

Reply:
If an error occurs getting the resource from the factory, or while trying to establish a connection, should we fail this whole process or just default back to READ_COMMITTED and proceed?

Reply:
They requested READ_UNCOMMITTED for a reason, which could lead to concurrent behaving unexpectedly or erroneously.  Without a FFDC or warning, the reason for this would be silent without trace enabled, and users are not likely going to have trace enabled until asked to enable it by support.  If there's a warning or FFDC, then they can at least bring that up at the very beginning when they engage support.

Reply:
While defaulting to READ_COMMITTED might be convenient for the particular scenario where persistent executor is using the persistence service with Oracle, behavior of the persistence service needs to be generally applicable for all components, and therefore an exception should go back to the caller.  Whether or not to FFDC as well should depend on whether we expect this to be something the caller will trap and handle another way.  If you have externalized the supportsIsolationLevel information to the caller and pattern is for the caller to create the READ_COMMITTED persistence unit, then ask it whether the database also supports READ_UNCOMMITTED then the caller has a way to know whether creation of the persistence unit with READ_UNCOMMITTED is valid or not before attempting it, in which case I would say go ahead and log an FFDC because they shouldn't be attempting this.  If, however, no provision is made for the caller to be able to detect whether the isolation level is supported, such that attempting to create the persistence unit and trapping for an exception _is_ the mechanism, then do not log FFDC because the exception is part of the expected programming model.  Sorry  that my answer is conditional, but github has marked this comment as outdated and will not place it in the code so I can't really see which way you have chosen or are planning to choose here.